keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -452,94 +462,200 @@ change are understandable. This may include API specs (though not always
+
 ![Privileged Container Diagram](Privileged.png)
 
 #### Networking
 - The container will be in the host’s network namespace (default network compartment) so it will have access to all the host’s network interfaces and have the host's IP as well.
+
 #### Resource Limits
 - Resource limits (disk, memory, cpu count) will be applied to the job and will be job wide. For example, with a limit of 10 MB is set for the job, if every process in the jobs memory allocations added up exceeds 10 MB this limit would be reached. This is the same behavior as other Windows container types. These limits would be specified the same way they are currently for whatever orchestrator/runtime is being used.
+
 #### Container Lifecycle 
 - The container's lifecycle will be managed by the container runtime just like other Windows container types.
-#### Container users 
--  The privileged container should be able to run as any user that's available on the host or in the domain of the host machine. Password accounts are being investigated.
+
+#### Container users
+
+- The privileged container can run as any user that's available on the host or in the domain of the host machine. Password accounts are being investigated.
+Running privileged containers as non SYSTEM/admin accounts will be the primary way operators can restrict access to system resources (files, registry, named pipes, WMI, etc).
+More information on Windows resource access can be found at https://docs.microsoft.com/en-us/archive/msdn-magazine/2008/november/access-control-understanding-windows-file-and-registry-permissions.
+
 #### Container Mounts
-- Directory mounts (i.e. directory C:\test mapped to C:\test in the privileged container) are still being investigated. Note that file system mapping in the current implementation would expose both the host filesystem and the container filesystem to the processes in the privileged container. We are still investigation how external mounts such as secrets or storage accounts would be added. The default visibility of all the host filesystem may present backwards compatibility challenges in the future if host path mounting is enabled in Windows. Currently the only method to restrict would be via the user in a way similar to [UID or GID](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) in the PSP. Currently however, this host path mounting is unlikely to be enabled in the future as a restriction of Job Objects.  
+
+- When privileged containers are started a new volume will be created on the host which will contain the contents of the container image. Containers will have a default working directory that points to this container volume. Containers will also have full access to the host file-system (unless restricted by filed-based ACLs and the run_as_username used to start the container.) Processes should use absolute paths when accessing files on the host and relative paths when accessing files brought in via the container image.
```

Comment:
By host filesystem, does that also include non-primary volumes from attachable disks? Also, after a container is started, would it see any new disks and filesystems that may have been attached afterwards?

Reply:
Yup, these "containers" are run as process (more details in #Overview) on the host and will be able to see **any** disks/volumes on the host file-system as they get mounted/unmounted.
Access to these new disks can be restricted via run_as_username functionality mentioned elsewhere in the KEP.

Reply:
In terms of access to other disks/volumes, I'm not concerned as privileged Linux containers today get access to all host devices.

I think though the requirement of using relative path or the env variable prefix will be the source of a lot of bugs involving accidental use/modifications of host files.

Reply:
Fortunately it is much more likely folks will be using relative paths (especially with cross-compiled golang binaries) which should help mitigate some risk.
Unfortunately we don't have any alternatives here after 3-4 months of designing/prototyping.

We're going to provide a lot of documentation around how these 'hostProcess' containers differ from 'privileged' container. Our hopes are that will be sufficient. 

Thoughts?

Reply:
I'm not familiar with the prevalence of workloads that use relative paths vs absolute, nor how common workloads are with fixed/unconfigurable paths. Although it seems like most of the targeted workloads for this feature are in the Kubernetes ecosystem so adjusting them should not be a major issue.

I'm wondering if we can do something in the API to make it very obvious that the volumeMount.mountPath is not going to show up the same in the container. For example, require the mountPath to be relative.

Reply:
aren't mount paths required to be absolute today? I don't think we'd want to ripple validation changes to other fields for this

Reply:
True, I'm mainly trying to see if there's a better way we could surface this semantic difference more obviously. We have had issues in the past where the [runtime didn't respect the mountPath](https://github.com/kubernetes/kubernetes/issues/85624) and that led to users writing to the host disk when they intended an external volume.

Reply:
To clarify, I don't think this is an alpha blocker, but we should definitely think about improving the UX for beta.

Reply:
> We have had issues in the past where the runtime didn't respect the mountPath and that led to users writing to the host disk when they intended an external volume.

yeah... ambiguous paths combined with high privilege is definitely disconcerting

Reply:
Just to clarify https://github.com/kubernetes/enhancements/pull/2288#discussion_r573382121, agree make this point more obvious for user is needed.

Reply:
Make this more obvious in the KEP or commit to heavily documenting this (which we definitely plan on doing)?

Reply:
KEP for alpha is good, for beta, consider better way we could surface this semantic difference more obviously as @msau42 mentioned.